### DISCUSSION TITLE
Onchain funded channel

### DISCUSSION THREAD

**Discussion by gzuuus** [ACINQ/phoenixd#24] [Q&A] [answered]
Can I open a channel using onchain funds, or can I only create a channel using lightning funds?

**Comment by pm47** [comment]
We call this feature "swap-in", currently it's not available in phoenixd (due to using mempool.space api instead of electrum for accessing the blockchain). We thought it was probably not going to be very useful for phoenixd users but can add it if needed.

**Reply by gzuuus** [reply]
It can be interesting feature to start using phoenixd and open a channel, otherwise the user needs to have the funds in ln beforehand and if you want to have a large liquidity this can be a bit cumbersome. The other option is to use a swap service, to exchange onchain for ln and then open the channel with those funds, but this adds an extra step, and fees.

**Reply by pm47** [reply]
The new [fee-credit feature](https://phoenix.acinq.co/server/auto-liquidity#auto-liquidity) saves you from doing all that though?

**Reply by gzuuus** [reply]
How? you still have to fund the channel with lightning so you have to have funds in lightning to open a channel, don't you?

**Reply by pm47** [reply]
You don't need to open a channel beforehand or to manage liquidity at all.

> the user needs to have the funds in ln beforehand and if you want to have a large liquidity this can be a bit cumbersome

If you want to cost effectively receive payment totalling for a large total amount of funds on Lightning:
- on phoenix mobile: you would indeed need to fund your channel (via ln or on-chain), then request 10M sat liquidity
- on phoenix server: just start phoenix with `./phoenixd --auto-liquidity=10m` and that's it. The first payment(s) received will automatically create a channel with the requested amount of liquidity. If the first payments are tiny, the fee credit will act as a buffer (see [concrete examples](https://phoenix.acinq.co/server/auto-liquidity#examples))

Not only is it much easier, the phoenix server approach saves one or two on-chain transaction (and fees) compared to the mobile approach, depending on whether you preload with ln or on-chain.

**Reply by gzuuus** [reply]
I see, but I was thinking of the use case of using phoenixd and being able to spend the funds you receive from the start, for that you need a channel, because if you don't have a channel you can receive funds, but those funds are unusable as they remain as fee credit, right? so until you have enough fee credit to open a channel the funds you receive are unusable. It is only when you have a channel and have received funds that you have a balance to spend. Correct me if I'm wrong, I just want to understand how this works.
Thanks for your answers.

**Reply by ghost** [reply]
If I understand phoenixd correctly, the entire application relies on the new "Request liquidity" feature introduced in mobile Phoenix. So you can't open your own channel with ACINQ (now), you can only buy inbound liquidity which will open a channel with you (with size of requested liquidity specified in --auto-liquidity= parameter) and now you can receive (and with enough outbound liquidity send) Lightning sats. 
I hope this helps!

**Reply by pm47** [reply]
> until you have enough fee credit to open a channel the funds you receive are unusable

This always true, with or without the fee credit. You need to be able to cover the channel creation cost before sending Lightning payments. If you don't want or need inbound liquidity because you plan to mostly send, you can set `--auto-liquidity off`.

> the entire application relies on the new "Request liquidity" feature introduced in mobile Phoenix

Not exactly. It's a major and very useful feature of phoenixd for sure, but if your use case is, e.g. do a giveaway where you just send payments, you won't need it.

**Reply by ghost** [reply]
But still you need a channel to send sats, and if you can't open one with sending on-chain funds (like in mobile Phoenix), you need to use "Request liquidity" feature, right? Or you can open a channel, but only through sending sats from different Lightning wallet?

**Reply by killer21m** [reply]
I also noticed that PhoenixD doesn't allow swap from on-chain. So if I wanted to pay 2m sat LN Invoice it wouldn't make sense to use PhoenixD, as I would need to send 2m sats from another LN wallet I control before sending the invoicd from PhoenixD. Whereas with Phoenix mobile, I can send 2-3m on-chain to Phoenix and use that channel to pay my LN invoice.

**Reply by a-nevermind** [reply]
Came here to add my 2 cents to demand for this feature. Depositing via on-chain and ability to generate new on-chain addresses each time (as in the Phoenix app) would be terrific!

**Reply by lnbc1QWFyb24** [reply]
Yeah want to add my 2 sats to this that it would be great to be able to fund onchain to trigger a channel open to get started. We have our phoenixd hooked up to a lnurl server for zaps, but we cannot send a nostr event for a zap receipt until we have a channel opened and websocket payment notifications start being triggered.

**Reply by doc-hex** [reply]
+1 for this feature. Required for feature parity with mobile version, IMHO.

**Reply by IanCardosoP** [reply]
+1 sat for this

**Reply by a-nevermind** [reply]
@pm47 do you have any further thoughts on this feature?

**Reply by matthiasdebernardini** [reply]
can we please add this feature, its so useful

**Comment by pm47** [comment, marked as answer]
This feature has been released in https://github.com/ACINQ/phoenixd/releases/tag/v0.9.0.
